src/ResourceManager/Sql/Commands.Sql/help/Add-AzureRmSqlElasticJobStep.md

```diff
@@ -0,0 +1,495 @@
+ [-CommandText] <String> [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobObject] <AzureSqlElasticJobModel>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Object Parameter Set Using Job Object
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseObject] <AzureSqlDatabaseModel> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobObject] <AzureSqlElasticJobModel>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Id Parameter Set Using Job Object
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseResourceId] <String> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobObject] <AzureSqlElasticJobModel>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Resource Id Parameter Set
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobResourceId] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Object Parameter Set Using Job Resource Id
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseObject] <AzureSqlDatabaseModel> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobResourceId] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Id Parameter Set Using Job Resource Id
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
```

Comment:
Sorry I didn't notice this earlier, but we need -StepId as a parameter so that steps can be inserted (i.e. not added to the end). And then please document how inserting steps works (see https://github.com/azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/jobs.json#L2845 )

Reply:
Will include step id to add with help message as suggested:
```
The job step's index within the job. 
If not specified when creating the job step, it will be created as the last step. 
If not specified when updating the job step, the step id is not modified.
```